core/src/ast/types.rs

```diff
@@ -511,19 +569,25 @@ impl fmt::Display for TypeName {
     }
 }
 
-#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+impl fmt::Display for PathType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.path.fmt(f)
+    }
+}
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
 pub enum Lifetime {
     Static,
```

Comment:
Why do we make a distinction for `'static`?

Reply:
It won't participate in lifetime-tracking, but it still needs to be known so we codegen it right. We either maintain the distinction here or later on during lifetime tracking, easier to do so here.